viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -4,11 +4,24 @@ import { TraceServerConfigService } from '../common/trace-server-config';
 @injectable()
 export class TraceServerServiceImpl implements TraceServerConfigService {
 
-    async startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
-        spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+    startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
+        const server = spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+        return new Promise((resolve, reject) => {
+            server.on('message', () => resolve());
+            server.on('error', () => reject());
+        });
```

Comment:
With the latest update in the trace server which can be downloaded here (https://download.eclipse.org/tracecompass.incubator/trace-server/rcp/), the server will return a non-zero code if it cannot be started successfully, e.g. when server is already running, port is taken etc. So, you need to add server.on('exit') and reject it. I tried it and worked.

Reply:
I've added an exit listener as well, though it will not be fired if the server has first emitted output. It would be possible to do something like 
```
onOutput
   timeout(resolve); // leave time to reject if it exits near-immediately.
onExit
  reject; // immediately
```
such that exiting within a certain interval, even if output were produced before it, would count as a rejection, but that would be somewhat arbitrary. Still, willing to consider that if it seems like the best approach.